query-builder/src/main/java/com/datastax/oss/driver/internal/querybuilder/insert/DefaultInsert.java

```diff
@@ -95,6 +98,32 @@ public JsonInsert json(@NonNull BindMarker json) {
         keyspace, table, json, missingJsonBehavior, ImmutableMap.of(), timestamp, ifNotExists);
   }
 
+  @NonNull
+  @Override
+  public JsonInsert json(@Nullable Object value, @NonNull CodecRegistry codecRegistry) {
+    return new DefaultInsert(
+        keyspace,
+        table,
+        QueryBuilder.literal(value, (value == null) ? null : codecRegistry.codecFor(value)),
```

Comment:
We should catch `CodecNotFoundException` and rethrow with a more context-specific message, like we did in [JAVA-2150](https://datastax-oss.atlassian.net/browse/JAVA-2150). See `QueryBuilder.literal(Object, CodecRegistry)`.

Reply:
Good idea. But I see one problem here:
https://github.com/datastax/java-driver/pull/1194/files#diff-ec647efb880ba814afddcf79eaa95570R402
`@Nullable Object value` can be null but next, we are doing assert `value != null`.
Then there is a possiblity in this line:
https://github.com/datastax/java-driver/pull/1194/files#diff-ec647efb880ba814afddcf79eaa95570R412
that it will cause NPE, isn't it?
Shouldn't the logic be:
```
  throw new IllegalArgumentException(
          String.format(
              "Could not inline literal of type %s. "
                  + "This happens because the driver doesn't know how to map it to a CQL type. "
                  + "Try passing a TypeCodec or CodecRegistry to literal().",
              (value == null) ? null : value.getClass().getName()),
          e);`
```
I will make it llike this in the `json()` `DefaultInsert`. Let me know wdyt. I can update logic in a `QueryBuilder` accordingly

Reply:
`null` is a valid literal in some CQL queries, but `INSERT ... JSON null` is invalid. So we can mark the value as `@NonNull`.
(the only case against it was if a codec encoded `null` into something else than the CQL literal `null`, but that's too contrived to be realistic)